roachpb/data.go

```diff
@@ -436,6 +437,16 @@ func (v *Value) SetTime(t time.Time) {
 	v.setTag(ValueType_TIME)
 }
 
+// SetDuration encodes the specified duration value into the bytes field of the
+// receiver, sets the tag and clears the checksum.
+func (v *Value) SetDuration(t duration.Duration) {
+	// TODO(dan): BEFORE MERGE Figure out the actual number for encodingSizeOverestimate.
+	const encodingSizeOverestimate = 33
+	v.RawBytes = make([]byte, headerSize, headerSize+encodingSizeOverestimate)
+	v.RawBytes = encoding.EncodeDurationAscending(v.RawBytes, t)
```

Comment:
The encoding for `roachpb.Value` does not need to sort correctly, so you might be able to speed this up or make the encoding smaller.